### PR TITLE
Support lustre client

### PR DESCRIPTION
**Pull request by sjpb**
- Enables building an "extra" image including a lustre client.
- Lustre client is built from source so will have OFED support.
- Automatically selects interface for Lnet configuration based on MGS NID.
- Only supports tcp Lnets

### CODE REVIEW

**Comment by sjpb** [comment]
Testing image build: https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11180817601

**Comment by sjpb** [comment]
Test image build: https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11181281999

**Comment by sjpb** [comment]
Test image build: https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11181427999

File: ansible/roles/lustre/tasks/install.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  register: _lustre_dnf_build_packages
+ 
+- name: Clone lustre git repo
+  # https://git.whamcloud.com/?p=fs/lustre-release.git;a=summary
+  ansible.builtin.git:
+    repo: git://git.whamcloud.com/fs/lustre-release.git
+    dest: "{{ lustre_build_dir }}"
+    version: "{{ lustre_version }}"
+
+- name: Prepare for lustre configuration
```
Comment (sd109) [inline ansible/roles/lustre/tasks/install.yml:13]: Typo?
```suggestion
- name: Prepare lustre configuration
```

**Review by sd109 — APPROVED** [review]
LGTM

File: ansible/roles/lustre/tasks/install.yml (hunk 1 above)
Comment (sjpb) [reply ansible/roles/lustre/tasks/install.yml:13]: Nope, the next step actually does the configuration

**Comment by sjpb** [comment]
build: https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11291367324

**Comment by sjpb** [comment]
Tested it can build a lustre "extra" image using either RL9 or RL9-OFED CI images at this commit.

**Comment by sjpb** [comment]
Build to test we haven't broken CI: https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11501463626

**Comment by sjpb** [comment]
Build to test this hasn't broken CI: https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11501525129

**Comment by sjpb** [comment]
Build: https://github.com/stackhpc/ansible-slurm-appliance/actions/runs/11501605073

**Review by sd109 — APPROVED** [review]
LGTM

**Pull request merged**